alpha/lib/model/Partner.php

```diff
@@ -2006,8 +2006,18 @@ public function getCdnHostWhiteList()
 
 	public function setCdnHostWhiteList($whiteListRegEx)
 	{
-		$whiteListArr = explode(',', rtrim($whiteListRegEx, ','));
-		$this->putInCustomData(self::CDN_HOST_WHITE_LIST, serialize($whiteListArr));
+		// set default value to null to avoid a case where we pass ' ' or '' and store an array of empty string
+		$value = null;
+		
+		// in case white space is given, store null
+		$whiteListRegEx = trim($whiteListRegEx);
+		if (!empty($whiteListRegEx))
+		{
+			$whiteListArr = explode(',', rtrim($whiteListRegEx, ','));
+			$value = serialize($whiteListArr);
+		}
+		
+		$this->putInCustomData(self::CDN_HOST_WHITE_LIST, $value);
```

Comment:
@yossipapi - this alone is enough to fix the issue
So far, this function always stored an array
Issue is when we tried to 'clear' this value which can be:
1. Via admin console partner config (see explanation below KalturaSystemPartnerConfiguration.php)
2. Via alpha - if we do $partner->setCdnHostWhiteList(''); in attempt to clear the value, we ended up storing an empty regex '' which preg_match always translats to true - leading to us building the domain+port at:
https://github.com/kaltura/server/blob/d0cca63de262f88bf308fee6fc8c932b110ae15c/alpha/apps/kaltura/lib/myPartnerUtils.class.php#L365-L364